### ISSUE TITLE
Calendar component clearing using model binding

### ISSUE THREAD

**Issue by odedolive** [primefaces/primeng#1840]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case**

Here's a demo of the issue:
http://plnkr.co/edit/LCUlngAPV77bkZHxd0gi?p=preview

**Current behavior**
When clearing a value that was entered manually by setting the bound date to null, the text in the calendar textbox is not getting cleared

**Expected behavior**
Value in textbox should be cleared the same way it gets cleared when the value in the textbox is set by selecting a value from the pop-out calendar

**Minimal reproduction of the problem with instructions**
All the info is in the plunker sample

**Comment by ribizli** [comment]
Confiming: this issue is there still in the 2.0.0 version

**Comment by ribizli** [comment]
I'm also trying to debug the issue. I think the problem is, that `writeValue` should directly modify the DOM, not expecting that changing `inputFieldValue`will be reflected in the view.
If you see how angular team implemented the [`DefaultValueAccessor`](https://github.com/angular/angular/blob/master/modules/%40angular/forms/src/directives/default_value_accessor.ts#L45-L48): they also do explicit DOM update:
```
this._renderer.setElementProperty(this._elementRef.nativeElement, 'value', normalizedValue);
```
The other solution would be using `ngModule`:
```
<input #inputfield type="text" pInputText [(ngModel)]="inputFieldValue" ...
```

Hope this analyse can help.

**Comment by ribizli** [comment]
For others suffering from this bug, here is my patch which handles the situation:
```
if (Calendar.prototype.onInputFocus) {
  const calendar = Calendar.prototype;
  let oldOnInputFocus = calendar.onInputFocus;
  calendar.onInputFocus = function(this: Calendar, inputfield, $event) {
    this['$$inputfield'] = inputfield;
    oldOnInputFocus.call(this, inputfield, $event);
  }

  let oldWriteValue = calendar.writeValue;
  calendar.writeValue = function(this: Calendar, v) {
    oldWriteValue.call(this, v);
    if (this['$$inputfield']) {
      this['$$inputfield'].value = this.inputFieldValue;
    }
  }
}
```

**Comment by ribizli** [comment]
This issue is fixed since version `2.0.4`

**Comment by cagataycivici** [comment]
This should be fixed in recent PrimeNG release, please try with it and if the issue persists drop a comment with along with a test case based on plunkr below and we'll review again.

http://plnkr.co/edit/Wj39h1?p=preview
